scalyr_agent/log_processing.py

```diff
@@ -1399,6 +1399,12 @@ def __init__(self, file_path, config, log_config, close_when_staleness_exceeds=N
         # Tracks whether the processor has recently logged data
         self.__is_active = False
 
+        # Tracks whether or not the processor has been marked as finished
+        self.__is_finished = False
```

Comment:
This means we have two pieces of state `__is_closed` and `__is_finished` that feels pretty similar to one another.  In effect, I don't think they mean anything different... when that state is `true`, the `LogMatcher` that owns it should remove it from its list of processors.  Is that correct, or am I missing something?

I have a strong preference for just re-using the concept of `close` for this feature.  Currently, a LogProcessor is considered `closed` when either (1) the file it was copying has been deleted or (2) the log file is considered stale because it has not had bytes written to it in X seconds and we are configured to close stale log files.  This would just introduce a new condition for that `close` case.

We do still need some additional state to record when a `LogProcessor` has been requested to close.  But, I think that's the only state we need.  We could call it `__close_when_caught_up`.  I'll add more down below to describe this.

Reply:
As discussed previously, LogMatcher still uses `finish`, but relies on LogProcesser being closed.